pingcap/dm
# *: log relay ignore reason; fix nil pointer for zap field

### What problem does this PR solve? 

1.  add the ignore reason for relay transformer/writer
2. fix a `nil` pointer used with `zap.Stringer`

### What is changed and how it works?

1. when ignoring a binlog event, attach a reason for the _result_ object
2. fix `nil` test for `interface`
```log
[2019/08/30 15:50:12.998 +08:00] [INFO] [relay.go:123] ["current earliest active relay log"] [task=test-2] [unit="binlog replication"] ["active relay logError"="PANIC=runtime error: invalid memory address or nil pointer dereference"]
```
```
[2019/08/30 22:56:01.965 +08:00] [INFO] [relay.go:123] ["current earliest active relay log"] [task=test-2] [unit="binlog replication"] ["active relay log"=NULL]
```

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
 	- start DM-worker, observe the log
```
[2019/08/30 22:55:10.026 +08:00] [INFO] [relay.go:388] ["ignore event by transformer"] [component="relay log"] [header="{\"Timestamp\":0,\"EventType\":27,\"ServerID\":11,\"EventSize\":39,\"LogPos\":11751,\"Flags\":0}"] [reason="heartbeat event"]
```
```
[2019/08/30 22:49:40.664 +08:00] [INFO] [relay.go:402] ["ignore event by writer"] [component="relay log"] [header="{\"Timestamp\":0,\"EventType\":4,\"ServerID\":11,\"EventSize\":47,\"LogPos\":0,\"Flags\":32}"] [reason="fake rotate event"]
```
